src/coreclr/jit/lsra.h

```diff
@@ -2062,17 +2064,30 @@ class LinearScan : public LinearScanInterface
     // NOTE: we currently don't need a LinearScan `this` pointer for this definition, and some callers
     // don't have one available, so make is static.
     //
-    static regMaskTP calleeSaveRegs(RegisterType rt)
+    static FORCEINLINE regMaskTP calleeSaveRegs(RegisterType rt)
     {
-        return varTypeIsIntegralOrI(rt) ? RBM_INT_CALLEE_SAVED : RBM_FLT_CALLEE_SAVED;
+        static const regMaskTP varTypeCalleeSaveRegs[] = {
+#define DEF_TP(tn, nm, jitType, sz, sze, asze, st, al, regTyp, regFld, csr, ctr, tf) csr,
+#include "typelist.h"
+#undef DEF_TP
+        };
+
+        assert((unsigned)rt < ArrLen(varTypeCalleeSaveRegs));
+        return varTypeCalleeSaveRegs[rt];
     }
 
+    // Not all of the callee trash values are constant, so don't declare this as a method local static
```

Comment:
not for this PR, but should we refactor this at other places as well?

Reply:
The other places are currently all constant, from what I saw at least. If we have any other non-constant places introduced in this PR then we should update them